src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Odnoklassniki
+{
+    public class OdnoklassnikiAuthenticationHandler : OAuthHandler<OdnoklassnikiAuthenticationOptions>
+    {
+        public OdnoklassnikiAuthenticationHandler(HttpClient client)
+            : base(client)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
+        {
+            var address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
+
+            // Signing.
+            // Call API methods using access_token instead of session_key parameter
+            // Calculate every request signature parameter sig using a little bit different way described in
+            // https://apiok.ru/dev/methods/
+            // session_secret_key = MD5(access_token + application_secret_key)
+            // sig = MD5(request_params_composed_string + session_secret_key)
+            // Don't include access_token into request_params_composed_string
+            var args = new Dictionary<string, string>
```

Comment:
Please consider moving that to a separate private/protected method like what we do in the other providers that require this stupid signing thing.